cpp/Doxyfile

```diff
@@ -38,7 +38,7 @@ PROJECT_NAME           = "WholeGraph C API"
 # could be handy for archiving the generated documentation or if some version
 # control system is used.
 
-PROJECT_NUMBER         = 25.02
+PROJECT_NUMBER         = $(RAPIDS_VERSION_MAJOR_MINOR)
```

Comment:
This is what other RAPIDS projects are doing, e.g. `rmm`: https://github.com/rapidsai/rmm/blob/8275ba8dc94a7fc63d9513acd81e3a2cf30a85d1/doxygen/Doxyfile#L41

One less hard-coded version is nice for:

* smaller diffs on updates when development switches to new versions
* simpler `update-version.sh`
* reduced risk of inconsistencies between docs and code